src/databricks/sqlalchemy/dialect/base.py

```diff
@@ -0,0 +1,17 @@
+import re
+from sqlalchemy.sql import compiler
+
+
+class DatabricksIdentifierPreparer(compiler.IdentifierPreparer):
+    # SparkSQL identifier specification:
+    # ref: https://spark.apache.org/docs/latest/sql-ref-identifier.html
+
+    legal_characters = re.compile(r"^[A-Z0-9_]+$", re.I)
```

Comment:
I thought that if you quote things get more lax than this

Reply:
Are these column or relation identifiers?

Reply:
Same rules apply to columns and relations. I think we borrowed this from pyhive. I will see about a test to prove exactly which character set to use.

Reply:
To expedite release, let's run with this character set for now. It's working in production for several customers / partners (we borrowed this regex from the PyHive dialect). If it becomes a limiting factor later then we can address with proper tests.